### ISSUE TITLE
Default column order doesn't have have any effect (`$cfg['Order']`) since phpMyAdmin 4.2.0

### ISSUE THREAD

**Issue by nunoperalta** [phpmyadmin/phpmyadmin#16851]
Changed this option to "DESC":

![image](https://user-images.githubusercontent.com/487079/115996038-a59bcb80-a5d5-11eb-9f57-b6e31ddddd48.png)

Logged out, logged in,
Go to a table, click on a column to sort, but still defaulting to "ASC":

![image](https://user-images.githubusercontent.com/487079/115996069-c106d680-a5d5-11eb-9f01-eb731e8c1409.png)
![image](https://user-images.githubusercontent.com/487079/115996060-bba98c00-a5d5-11eb-8746-57042a31998a.png)

- Using Docker
- Database version: 10.5.9-MariaDB
- PHP version: 7.4.15
- phpMyAdmin version: 5.1.0

**Comment by williamdes** [comment]
Could you send a table structure please ?

**Comment by nunoperalta** [comment]
```sql
CREATE TABLE `test` (
  `tid` bigint(18) UNSIGNED NOT NULL,
  `ip` varchar(45) CHARACTER SET utf8 NOT NULL,
  `time` int(10) UNSIGNED NOT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci ROW_FORMAT=DYNAMIC;

ALTER TABLE `forums_topics_views`
  ADD PRIMARY KEY (`tid`,`ip`),
  ADD KEY `time` (`time`);
```

Click on the "time" column to sort. Defaults to ASC instead of DESC.

**Comment by williamdes** [comment]
> Click on the "time" column to sort. Defaults to ASC instead of DESC.

yeah but this is not the initial feature code, so this is more like something we would expect but was never thought about if I am right

Ref: https://github.com/phpmyadmin/phpmyadmin/issues/16832#issuecomment-826328774

**Comment by OlafvdSpek** [comment]
@williamdes $cfg['Order'] is something else than #16832 isn't it?

**Comment by williamdes** [comment]
> Order

Yes: https://docs.phpmyadmin.net/en/latest/config.html#cfg_Order

> I.E. descending order for columns of type TIME, DATE, DATETIME and TIMESTAMP, ascending order else- by default.

That would not apply to the `time` column, right ?

**Comment by williamdes** [comment]
@nunoperalta Did you open this issue for `$cfg['Order']` ?

**Comment by nunoperalta** [comment]
@williamdes You seem to be talking about this:

https://docs.phpmyadmin.net/en/latest/config.html#cfg_TablePrimaryKeyOrder

However, I'm talking about this:

https://docs.phpmyadmin.net/en/latest/config.html#cfg_Order

"Default column order", which is not for primary key.

I want to set a default sort for when I click ANY column to sort.

**Comment by nunoperalta** [comment]
> That would not apply to the time column, right ?

Not if I set "SMART"

But if I set "DESC", it should apply to any column, no matter what the data type is.

**Comment by williamdes** [comment]
> But if I set "DESC", it should apply to any column, no matter what the data type is.

Okay, well I will check the code and let you know if this is the original design so we can decide on what branch to fix it ;)

**Comment by williamdes** [comment]
Hi @nunoperalta 
Could you check #18130 and let me know if I understood your request about this setting ?
And that it covers 100% of this issue

**Comment by nunoperalta** [comment]
The "Default sorting order" config works well for the Primary Key default sort, when we open a table in phpMyAdmin.
I have "DESC" selected, and the Primary Key is defaulted to DESC when I open any table. That's fine 👍

What I am reporting in #16851 is that when I click on another column that is not the Primary Key (say `unix_time`, or `total_something`, instead of `id`), I always have to click 2 times, because the first time it will always sort ASC.

**Comment by williamdes** [comment]
> The "Default sorting order" config works well for the Primary Key default sort, when we open a table in phpMyAdmin. I have "DESC" selected, and the Primary Key is defaulted to DESC when I open any table. That's fine 👍
> 
> What I am reporting in #16851 is that when I click on another column that is not the Primary Key (say `unix_time`, or `total_something`, instead of `id`), I always have to click 2 times, because the first time it will always sort ASC.

Awesome, regardless of changing SMART to DESC.
In SMART mode it would be the feature request #18130

So I think I will publish a fix for what you reported and you will be able to change the default sort setting after that, like it worked 10 years ago

**Comment by williamdes** [comment]
Fixed by 7d4eafd188030307dd8666ed32cf26510466d1bc
